File: lessons/lesson-04-control-flow/js-compare/controlFlow.js
// JavaScript — Control Flow
// Run: node controlFlow.js

let score = 85;

// if / else if / else — SAME in Java
if (score >= 90) {
    console.log("Grade: A");
} else if (score >= 80) {
    console.log("Grade: B");
} else if (score >= 70) {
    console.log("Grade: C");
} else {
    console.log("Grade: F");
}

// Ternary — SAME in Java
let result = score >= 60 ? "Pass" : "Fail";
console.log(result);

// Switch — SAME concept
let day = "Saturday";
switch (day) {
    case "Monday":
    case "Tuesday":
    case "Wednesday":
    case "Thursday":
    case "Friday":
        console.log("Weekday");
        break;
    case "Saturday":
    case "Sunday":
        console.log("Weekend");
        break;
    default:
        console.log("Invalid day");
}

// Validation pattern
function processId(id, maxPlayers) {
    if (id < 1 || id > maxPlayers) {
        console.log("Invalid id: " + id);
        return; // early return
    }
    console.log("Processing player " + id);
}

processId(5, 11);  // valid
processId(15, 11); // invalid

File: lessons/lesson-04-control-flow/java-code/ControlFlow.java
// Java — Control Flow (almost identical to JS!)

public class ControlFlow {
    public static void main(String[] args) {

        int score = 85;

        // if / else if / else — literally the same as JS
        if (score >= 90) {
            System.out.println("Grade: A");
        } else if (score >= 80) {
            System.out.println("Grade: B");
        } else if (score >= 70) {
            System.out.println("Grade: C");
        } else {
            System.out.println("Grade: F");
        }

        // Ternary — same as JS
        String result = score >= 60 ? "Pass" : "Fail";
        System.out.println(result);

        // Switch — same concept, Java supports String since Java 7
        String day = "Saturday";
        switch (day) {
            case "Monday":
            case "Tuesday":
            case "Wednesday":
            case "Thursday":
            case "Friday":
                System.out.println("Weekday");
                break;
            case "Saturday":
            case "Sunday":
                System.out.println("Weekend");
                break;
            default:
                System.out.println("Invalid day");
        }
    }
}
